alica_engine/include/engine/constraintmodul/VariableSyncModule.h

```diff
@@ -66,4 +74,67 @@ class VariableSyncModule : public IVariableSyncModule {
     supplementary::NotifyTimer<VariableSyncModule>* _timer;
     mutable std::mutex _mutex;
 };
+
+template <typename VarType>
+int VariableSyncModule::getSeeds(const std::vector<VarType*>& query, const std::vector<Interval<double>>& limits, std::vector<Variant>& o_seeds) const
+{
+    const int dim = query.size();
+    // TODO: use only stack memory for low dimensionality
+    std::vector<Variant> vec(dim);
+
+    std::vector<VotedSeed> seeds;
+
+    AlicaTime earliest = _ae->getAlicaClock()->now() - _ttl4Usage;
+    if (_ownResults.getValues(query, earliest, vec)) {
+        seeds.emplace_back(std::move(vec));
+    }
+
+    std::lock_guard<std::mutex> lock(_mutex);
```

Comment:
could the scope of this lock be reduced?

Reply:
good catch